packages/wdio-cucumber-framework/src/wdioEventTranslator.js

```diff
@@ -0,0 +1,298 @@
+import { Status } from 'cucumber'
+import { CucumberEventListener } from './eventListener'
+import { createStepArgument } from './utils'
+import * as path from 'path'
+import logger from '@wdio/logger'
+
+const SETTLE_TIMEOUT = 5000
+const log = logger('wdio-cucumber-framework:CucumberWdioEventTranslator')
+
+/**
+ * CucumberReporter
+ * responsible for accepting the cucumber specific events emitted by CucumberEventListener
+ * and emitting the standard WebdriverIO test runtime events
+ * The translation is:
+ *
+ * before-feature/after-feature -> suite:start/suite:end
+ * before-scenario/after-scenario -> suite:start/suite:end
+ * before-step/after-step -> test:start/test:end
+ *
+ * The step == test translation doesn't map the semantics of cucumber very well,
+ * but does allow WebdriverIO to maintain behavior that's more consistent with Cucumber's
+ * expected behavior
+ *
+ * Note that if you are building Cucumber-specific functionality, you probably
+ * want to instead rely directly on the Cucumber-specific events
+ */
+// TODO - this class name isn't great, becuase it doesn't capture that this propogates the events
+// as well as translating
+class CucumberWdioEventTranslator {
```

Comment:
I think `CucumberReporter` is perfectly fine name, it may cause confusion if the `wdio-cucumber-reporter` so maybe `CucumberEventReporter` as it is a reporter it would translate and propagate

Reply:
I personally like CucumberEventReporter. I think that is enough to add clarity

Reply:
Works for me!